### PR TITLE
Reimplement more Schema Manager listTable*() methods with the new API

### PR DESCRIPTION
**Pull request by morozov**
After having merged the changes from https://github.com/doctrine/dbal/pull/5268 up to `4.0.x` and trying to remove the deprecated methods, I realized that some of the schema manager methods still use the old implementation.

It would make sense to rework them using the new API for consistency.

The fact that the `listTableColumns()`, `listTableIndexes()` and `listTableForeignKeys()` exist alongside `listTableDetails()` (which provides a super-set of the data provided by the first three methods) brings up the question: do those method need to exist as part of the Schema Manager API in the first place? I don't believe they are used directly by the ORM or Migrations for schema management but they are used by the test suite for introspection of the specific schema aspects (e.g. for testing introspection of the foreign keys).

For now, I propose to leave these methods in place but later, when we have a dedicated API for schema introspection, deprecate them in the schema manager. The test suite could use the lower-level API for more detailed introspection while the external consumers would continue using the higher-level API w/o the need for the low-level details.

### CODE REVIEW

File: src/Schema/SQLServerSchemaManager.php [hunk 1]
```diff
@@ -272,26 +296,6 @@ protected function _getPortableViewDefinition($view)
...
-     * {@inheritdoc}
-     */
-    public function listTableIndexes($table)
-    {
-        $sql = $this->_platform->getListTableIndexesSQL($table, $this->_conn->getDatabase());
-
-        try {
-            $tableIndexes = $this->_conn->fetchAllAssociative($sql);
-        } catch (Exception $e) {
-            if (strpos($e->getMessage(), 'SQLSTATE [01000, 15472]') === 0) {
```
Comment (morozov) [inline src/Schema/SQLServerSchemaManager.php:285]: This logic was added in 03be671e1df6338a7b528646fb51bea537dd44b7. Judging by the build, it's not necessary any longer or isn't properly tested.

File: tests/Schema/MySQLSchemaManagerTest.php [hunk 2]
```diff
@@ -1,98 +0,0 @@
...
-use Doctrine\DBAL\Platforms\AbstractMySQLPlatform;
-use Doctrine\DBAL\Schema\AbstractSchemaManager;
-use Doctrine\DBAL\Schema\ForeignKeyConstraint;
-use Doctrine\DBAL\Schema\MySQLSchemaManager;
-use PHPUnit\Framework\MockObject\MockObject;
-use PHPUnit\Framework\TestCase;
-
-use function array_map;
-
-class MySQLSchemaManagerTest extends TestCase
```
Comment (morozov) [inline tests/Schema/MySQLSchemaManagerTest.php:18]: This test covers the implementation details that have changed as of this PR. It was introduced in https://github.com/doctrine/dbal/pull/97 alongside an integration test that still passes.

File: src/Schema/AbstractSchemaManager.php [hunk 3]
```diff
@@ -211,6 +211,29 @@ public function listTableColumns($table, $database = null)
         return $this->_getPortableTableColumnList($table, $database, $tableColumns);
     }
 
+    /**
+     * @param string      $table
+     * @param string|null $database
+     *
+     * @return Column[]
```
Comment (greg0ire) [inline src/Schema/AbstractSchemaManager.php:218]: Why not use native type declarations for this?

Comment (morozov) [reply src/Schema/AbstractSchemaManager.php:218]: For return types, yes, makes sense.

**Review by greg0ire — APPROVED** [review]

**Pull request merged**